common/parallel/FileSystemCrawler.go

```diff
@@ -96,7 +98,11 @@ func Walk(root string, parallelism int, parallelStat bool, walkFn filepath.WalkF
 	// walk the stuff inside the root
 	reader, remainingParallelism := NewDirReader(parallelism, parallelStat)
 	defer reader.Close()
-	ctx, cancel := context.WithCancel(context.Background())
+	if appCtx != nil {
+		ctx, cancel = context.WithCancel(appCtx)
```

Comment:
after changing name to ctx this will simply be

ctx, cancel := context.WithCancel(ctx)

Reply:
I used appCtx to tell its application context. So that code more readable.